pythonfmu/csvbuilder.py

```diff
@@ -1,30 +1,68 @@
 import argparse
 import tempfile
 from pathlib import Path
-from typing import Union, Optional
+from typing import Union
 from ._version import __version__
 from .fmi2slave import FMI2_MODEL_OPTIONS
 from .builder import FmuBuilder
 
 FilePath = Union[str, Path]
 
 
-def create_csv_slave(csvfile: FilePath):
-    classname = csvfile.stem
-    filename = csvfile.name
+def create_csv_slave(csv_file: FilePath):
+    classname = csv_file.stem
+    filename = csv_file.name
     return f"""
-from pythonfmu.fmi2slave import Fmi2Slave, Fmi2Causality, Fmi2Variability, Real
+import re
 import csv
-
-EPS = 1e-6
+from math import isclose  # requires >= python 3.5
+from pythonfmu.fmi2slave import Fmi2Type, Fmi2Slave, Fmi2Causality, Fmi2Variability, Integer, Real, Boolean, String
 
 def lerp(v0: float, v1: float, t: float) -> float:
     return (1 - t) * v0 + t * v1
 
-
 def normalize(x, in_min, in_max, out_min, out_max):
+    x = max(min(x, in_max), in_min)
+    # print(f"x={{x}}, in_min={{in_min}}, in_max={{in_max}}, out_min={{out_min}}, out_max={{out_max}}, ")
     return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min
 
+def get_fmi2_type(s: str) -> Fmi2Type:
+  s_lower = s.lower()
+  if Fmi2Type.integer.name in s_lower:
+    return Fmi2Type.integer
+  elif Fmi2Type.real.name in s_lower:
+    return Fmi2Type.real
+  elif Fmi2Type.boolean.name in s_lower:
+    return Fmi2Type.boolean
+  elif Fmi2Type.string.name in s_lower:
+    return Fmi2Type.string
+  elif Fmi2Type.enumeration.name in s_lower:
+    raise TypeError("Unsupported type: {{Fmi2Type.enumeration.name}}")
+  else:
+    raise TypeError("Could not process type from input string: {{s}}")
```

Comment:
What about this version? Note: I think you should be more stringent how the format to provide the type. Checking for `type in` is too broad. It is better to force the type to `startswith` or `endswith`.

```suggestion
def get_fmi2_type(s: str) -> Fmi2Type:
    s_lower = s.lower()
    for type in Fmi2Type:
        if type.name in s_lower:
            if type == Fmi2Type.enumeration:
                raise TypeError("Unsupported type: {{Fmi2Type.enumeration.name}}")
            else:
                return type
    raise TypeError("Could not process type from input string: {{s}}")
```

Reply:
It has to do with headers generated from a specific tool we use.